x-pack/plugins/observability_solution/investigate_app/public/pages/details/investigation_details_page.tsx

```diff
@@ -22,8 +26,46 @@ export function InvestigationDetailsPage() {
     },
   } = useKibana();
 
+  const {
+    path: { id },
+  } = useInvestigateParams('/{id}');
+
   const ObservabilityPageTemplate = observabilityShared.navigation.PageTemplate;
 
+  const {
+    data: investigationDetails,
+    isLoading: isFetchInvestigationLoading,
+    isError: isFetchInvestigationError,
+  } = useFetchInvestigation({ id });
+
+  const alertId = investigationDetails?.origin.id ?? '';
+
+  const {
+    data: alertDetails,
+    isLoading: isFetchAlertLoading,
+    isError: isFetchAlertError,
+  } = useFetchAlert({ id: alertId });
```

Comment:
More of a philosophical question, but do you think the get investigation API should query RAC and include the alert details directly in its response?
So instead of doing two request sequentially, the client would just do one and get everything?
That's maybe a bit of a premature optimization. But something to think about.

Reply:
that's a good question, something I was thinking about it too while implementing. We would need to have `rac` in context to use alerts client. I will look into it.